include/aspect/utilities.h

```diff
@@ -197,6 +197,18 @@ namespace aspect
     signed_distance_to_polygon(const std::vector<Point<2> > &point_list,
                                const dealii::Point<2> &point);
 
+
+    /**
+    * Given a 2d point and a list of two points that define a line, compute the smallest
+    * distance of the point to the line segment. Note that when the point's perpendicular
+    * base does not lie on the line segment, the smallest distance to the segment's end
+    * points is calculated.
+    */
+    template <int dim>
+    double
+    distance_to_line(const std::vector<dealii::Point<2> > &point_list,
```

Comment:
Since you know that it's only two lines, why not use a `std::array<Point<2>,2>`?

Reply:
Also, nothing in this function depends on the template argument `dim` -- I think you can just make it a non-template.

Unless, of course, you want to generalize this to general line segments in 3d as well :-)